src/thunderbots/software/test/util/time/duration.cpp

```diff
@@ -0,0 +1,123 @@
+#include "util/time/duration.h"
+
+#include <gtest/gtest.h>
+
+#include <stdexcept>
+
+TEST(DurationTest, default_constructor)
+{
+    Duration d;
+    EXPECT_DOUBLE_EQ(d.getSeconds(), 0);
+}
+
+TEST(DurationTest, create_duration_from_positive_seconds)
+{
+    Duration duration = Duration::fromSeconds(1.0);
+    EXPECT_DOUBLE_EQ(duration.getSeconds(), 1.0);
+}
+
+TEST(DurationTest, create_duration_from_negative_seconds)
+{
+    Duration duration = Duration::fromSeconds(-0.2);
+    EXPECT_DOUBLE_EQ(duration.getSeconds(), -0.2);
+}
+
+TEST(DurationTest, create_duration_from_zero_milliseconds)
+{
+    Duration duration = Duration::fromMilliseconds(0);
+    EXPECT_DOUBLE_EQ(duration.getMilliseconds(), 0);
+}
+
+TEST(DurationTest, create_duration_from_negative_milliseconds)
+{
+    Duration duration = Duration::fromMilliseconds(-125);
+    EXPECT_DOUBLE_EQ(duration.getMilliseconds(), -125);
+}
+
+TEST(DurationTest, get_seconds_duration_in_milliseconds)
```

Comment:
Missing test case for `get_seconds_duration_in_seconds`?

Reply:
added